lib/core/plugin.js

```diff
@@ -119,43 +121,45 @@ Plugin.prototype.interceptLogs = function(context) {
 Plugin.prototype.registerClientWeb3Provider = function(cb) {
   this.clientWeb3Providers.push(cb);
   this.pluginTypes.push('clientWeb3Provider');
-};
-
-Plugin.prototype.registerBeforeDeploy = function(cb) {
-  this.beforeDeploy.push(cb);
-  this.pluginTypes.push('beforeDeploy');
+  this.pluginTypes = _.uniq(this.pluginTypes);
```

Comment:
Crazy idea, you could override `this.pluginTypes.push()` to trigger a _.uniq after